src/config.rs

```diff
@@ -0,0 +1,177 @@
+        self.bindings.get(&code).copied()
+    }
+
+    pub fn get_keycode_string(&self, action: Action) -> String {
+        match self.reverse.get(&action) {
+            Some(KeyCode::Char(c)) => format!("[{}]", c),
+            Some(other) => format!("[{:?}]", other),
+            None => "[???]".to_string(),
+        }
+    }
+}
+
+fn default_keybindings() -> Keymap {
+    let mut bindings = HashMap::new();
+
+    bindings.insert(KeyCode::Char('k'), Action::Up);
+    bindings.insert(KeyCode::Char('j'), Action::Down);
+    bindings.insert(KeyCode::Char('h'), Action::Left);
+    bindings.insert(KeyCode::Char('l'), Action::Right);
+    bindings.insert(KeyCode::Char('u'), Action::PageUp);
+    bindings.insert(KeyCode::Char('d'), Action::PageDown);
+    bindings.insert(KeyCode::Char('q'), Action::Quit);
+    bindings.insert(KeyCode::Esc, Action::Reset);
+    bindings.insert(KeyCode::Tab, Action::NextTab);
+    bindings.insert(KeyCode::BackTab, Action::PrevTab);
+
+    let reverse = bindings
+        .iter()
+        .map(|(k, v)| (*v, *k))
+        .collect::<HashMap<_, _>>();
+
+    Keymap { bindings, reverse }
+}
+
+fn parse_keycode(s: &str) -> Result<KeyCode, String> {
+    Ok(match s {
+        // single characters
+        c if c.len() == 1 => KeyCode::Char(c.chars().next().unwrap()),
+
+        "Backspace" => KeyCode::Backspace,
+        "Enter" => KeyCode::Enter,
+        "Left" => KeyCode::Left,
+        "Right" => KeyCode::Right,
+        "Up" => KeyCode::Up,
+        "Down" => KeyCode::Down,
+        "Home" => KeyCode::Home,
+        "End" => KeyCode::End,
+        "PageUp" => KeyCode::PageUp,
+        "PageDown" => KeyCode::PageDown,
+        "Tab" => KeyCode::Tab,
+        "BackTab" => KeyCode::BackTab,
+        "Delete" => KeyCode::Delete,
+        "Insert" => KeyCode::Insert,
+        "Esc" | "Escape" => KeyCode::Esc,
+
+        // Function keys
+        k if k.starts_with("F") => {
+            let n = k[1..]
+                .parse::<u8>()
+                .map_err(|_| format!("Bad function key: {}", k))?;
+            KeyCode::F(n)
+        }
+
+        _ => return Err(format!("Unknown key '{}'", s)),
+    })
+}
+
+fn build_keymap(raw: RawConfig) -> Keymap {
+    let mut bindings = default_keybindings().bindings;
+
+    for (key_str, action) in raw.keybindings {
+        match parse_keycode(&key_str) {
+            Ok(code) => {
+                // Remove any existing key that maps to the same action
+                bindings.retain(|_, &mut a| a != action);
+
+                // Insert the new key
+                bindings.insert(code, action);
+            }
+            Err(err) => eprintln!("[config] bad key '{}': {} — ignored", key_str, err),
+        }
+    }
+
+    let reverse = bindings
+        .iter()
+        .map(|(k, v)| (*v, *k))
+        .collect::<HashMap<_, _>>();
+
+    Keymap { bindings, reverse }
+}
+
+pub fn load_config() -> AppConfig {
+    let proj_dirs = ProjectDirs::from("", "", "parqeye").unwrap();
```

Comment:
In macOS, this sets the path to [`.../Library/Application Support/...`](https://docs.rs/directories/latest/directories/struct.ProjectDirs.html#examples). The config should live in the same path as in Linux (`$HOME/.config/parqeye`), which a few other command-line tools resolve to in macOS. [This](https://github.com/lavifb/bat/blob/4a53b95ce6d05b5535e40bcc575a8ab90d3f8cf2/src/dirs.rs#L28-L35) might be the best solution I see for resolving paths.